sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/OpenTelemetryTracer.java

```diff
@@ -350,6 +349,23 @@ private Builder getSpanBuilder(String spanName, Context context) {
         return TRACER.spanBuilder(spanNameKey).setParent(parentSpan);
     }
 
+    /**
+     * Extracts a {@link Span.Builder} from the given {@code context}.
+     *
+     * @param context The context containing the Span.Builder.
+     * @return The {@link Span.Builder} contained in the context, and {@code null} if it does not.
+     */
+    private Span.Builder getSharedSpanBuilder(Context context) {
+        final Object value = getOptionalObject(context, SPAN_BUILDER_KEY);
```

Comment:
I think `getOptionalObject` is clunky. You have to perform the same check over for an instance every time you get from it. I'm not sure what work it is actually saving you.

```java
    @SuppressWarnings("unchecked")
    public <T> T getOrDefault(Object key, T defaultValue, Class<T> clazz) {
        final Optional<Object> optional = getData(key);
        final Object result = optional.filter(value -> value.getClass().equals(clazz)).orElse(() -> {
           logger.warning("Could not extract key '{}' of type '{}' from context.", key, clazz);
           return defaultValue;
        });

        return (T) result;
    }
```

Reply:
I think this method helps to clear out a lot of redundant methods, thanks @conniey 🎆